### PR TITLE
Add static analysis pyright test to riot

### PR DESCRIPTION
**Pull request by bojbrook**
## Description
Includes a new static analysis test using [pyright](https://github.com/microsoft/pyright) that can be ran using riot.

These are the most common error being thrown along with an example log to go with it.
- **TYPE Import errors** `error: Import "ddtrace.profiling" could not be resolved`
- **Unsupported Type operator** `/root/project/ddtrace/internal/atexit.py:56:76 - error: Operator "!=" not supported for types "(*args: Any, **kwargs: Any) -> None" and "(*args: Any, **kwargs: Any) -> None"`
- **Type ISSUES**: `error: No parameter named <parameter>`
- **UNKNOWN SYMBOL error**: `<> is unknown import symbol`
- **Cannot access Member** `/root/project/ddtrace/internal/runtime/runtime_metrics.py:34:41 - error: Cannot access member "ENABLED" for type "RuntimeCollectorsIterable" Member "ENABLED" is unknown (reportGeneralTypeIssues)`

The errors fall under the following diagnostics rules which can be configured to be ignored the pyrightconfig.json file - https://github.com/microsoft/pyright/blob/main/docs/configuration.md#diagnostic-rule-defaults 
- reportGeneralTypeIssues
- reportMissingImports
- reportPrivateImportUsage
- reportOptionalMemberAccess

I'm happy to delve further into this static analysis configuration if more research is required.

### CODE REVIEW

File: pyrightconfig.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  "exclude": [
+    "**/node_modules",
+    "**/__pycache__",
+    ".git",
+    "ddtrace/vendor",
+    "ddtrace/contrib"
+  ],
+  "executionEnvironments": [
+    {
+      "root": "ddtrace"
```
Comment (brettlangdon) [inline pyrightconfig.json:12]: I think we want this as `.` instead of `ddtrace`

**Comment by brettlangdon** [comment]
Can we also add this to the `pre_check` CircleCI job so we can test/validate the errors together?

https://github.com/DataDog/dd-trace-py/blob/2acfeabba1d7ee16f6d713b118753d1493350581/.circleci/config.yml#L278-L294

we'll want the command to also install pyright via npm until the docker file gets updated.

File: pyrightconfig.json (hunk 1 above)
Comment (bojbrook) [reply pyrightconfig.json:12]: This resolved the import issues

**Comment by bojbrook** [comment]
Adding pyright to the circleCI precheck wasn't working since the the docker image `python39` doesn't have npm installed. Instead I have added all the errors into this [gist(dd-trace-py pyright output)](https://gist.github.com/bojbrook/9aceff42f0eb4630358178e89d472758) to be reviewed by the team.

**Comment by P403n1x87** [comment]
Should this also run as part of the pre-commit verification hooks?

**Comment by brettlangdon** [comment]
Closing this for now since we don't have the bandwidth to bring it across the finish line, but thanks @bojbrook for the investigation, really helpful!

**Pull request closed without merging**